tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/MetricName.java

```diff
@@ -39,6 +39,17 @@
     @Value.NaturalOrder
     SortedMap<String, String> safeTags();
 
+    /**
+     * Creates a metric name with the specified extra tag name and value entry.
+     * <p>All tags and keys must be {@link Safe} to log.
+     * @param extraTagName extra tag key name
+     * @param extraTagValue extra tag value
+     * @return metric name including keys
+     */
+    default MetricName withExtraTag(@Safe String extraTagName, @Safe String extraTagValue) {
+        return RealMetricName.create(this, extraTagName, extraTagValue);
```

Comment:
The primary consumer of this will likely be [Dialogue's `VersionedTaggedMetricRegistry#augment`](https://github.com/palantir/dialogue/blob/a7cc75d8203de4189ad10a95c59cf8d187f5dc23/dialogue-core/src/main/java/com/palantir/dialogue/core/VersionedTaggedMetricRegistry.java#L51-L56)

Reply:
Dialogue case handled by https://github.com/palantir/dialogue/pull/1609